packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -1156,8 +1162,43 @@ export class CdkToolkit {
       defaultBehavior: DefaultSelection.OnlySingle,
     });
 
-    // No validation
+    const selectorWithoutPatterns: StackSelector = {
+      ...selector,
+      allTopLevel: true,
+      patterns: [],
+    };
+    const stacksWithoutPatterns = await assembly.selectStacks(selectorWithoutPatterns, {
+      extend: exclusively ? ExtendedStackSelection.None : ExtendedStackSelection.Downstream,
+      defaultBehavior: DefaultSelection.OnlySingle,
+    });
+
+    const patterns = selector.patterns.map(pattern => {
+      const notExist = !stacks.stackArtifacts.find(stack =>
+        minimatch(stack.hierarchicalId, pattern) || (stack.id === pattern && semver.major(versionNumber()) < 2),
+      );
+
+      const closelyMatched = notExist ? stacksWithoutPatterns.stackArtifacts.map(stack => {
+        if (minimatch(stack.hierarchicalId.toLowerCase(), pattern.toLowerCase())) {
+          return stack.hierarchicalId;
+        }
+        if (stack.id.toLowerCase() === pattern.toLowerCase() && semver.major(versionNumber()) < 2) {
+          return stack.id;
+        }
```

Comment:
No need to check this, versionNumber is never < 2 anymore.

Reply:
Thanks, fixed:

https://github.com/aws/aws-cdk/pull/32636/commits/d52932428128eef7ebda60fa52dd250f7652d904